bitbots_quintic_walk/src/walk_pywrapper.cpp

```diff
@@ -0,0 +1,264 @@
+}
+
+PyWalkWrapper::PyWalkWrapper(const std::string ns) : walk_node_(std::make_shared<bitbots_quintic_walk::WalkNode>(ns)) {
+  //walk_node_.reset(new );
+  //ros::init(empty, "walking");
+  //
+  //walk_node_.reset(new bitbots_quintic_walk::WalkNode());
+  set_robot_state(0);
+}
+
+moveit::py_bindings_tools::ByteString PyWalkWrapper::step(double dt,
+                                                          const std::string &cmdvel_msg,
+                                                          const std::string &imu_msg,
+                                                          const std::string &jointstate_msg) {
+  std::string result = to_python<bitbots_msgs::JointCommand>(walk_node_->step(dt,
+                                                                              from_python<geometry_msgs::Twist>(
+                                                                                  cmdvel_msg),
+                                                                              from_python<sensor_msgs::Imu>(imu_msg),
+                                                                              from_python<sensor_msgs::JointState>(
+                                                                                  jointstate_msg)));
+  return moveit::py_bindings_tools::serializeMsg(result);
+}
+
+moveit::py_bindings_tools::ByteString PyWalkWrapper::get_left_foot_pose() {
+  std::string result = to_python<geometry_msgs::Pose>(walk_node_->get_left_foot_pose());
+  return moveit::py_bindings_tools::serializeMsg(result);
+}
+
+void PyWalkWrapper::reset() {
+  walk_node_->reset();
+}
+
+float PyWalkWrapper::get_phase() {
+  return walk_node_->getEngine()->getPhase();
+}
+
+float PyWalkWrapper::get_freq() {
+  return walk_node_->getEngine()->getFreq();
+}
+
+void PyWalkWrapper::set_robot_state(int state) {
+  humanoid_league_msgs::RobotControlState state_msg;
+  state_msg.state = state;
+  walk_node_->robStateCb(state_msg);
+}
+
+void PyWalkWrapper::set_engine_dyn_reconf(const boost::python::object params) {
+  using namespace boost::python;
+
+  extract<dict> cppdict_ext(params);
+  if (!cppdict_ext.check()) {
+    throw std::runtime_error(
+        "PassObj::pass_dict: type error: not a python dict.");
+  }
+
+  dict cppdict = cppdict_ext();
+  list keylist = cppdict.keys();
+
+  // create dyn reconf object
+  bitbots_quintic_walk::bitbots_quintic_walk_engine_paramsConfig dyn_conf;
+
+  // fill all values from dict to dyn reconf
+  int const len = boost::python::len(keylist);
+  //std::cout << "len(keylist) = " << len << std::endl;
+  // since c++ has no reflection we have to do this in a bad way
+  for (int i = 0; i < len; ++i) {
+    // operator[] is in python::boost::object
+    std::string keystr = extract<std::string>(str(keylist[i]));
+    std::string valstr = extract<std::string>(str(cppdict[keylist[i]]));
+    //std::cout << "key:[" << keystr << "]->[" << valstr << "]" << std::endl;
+    if (keystr == "double_support_ratio") {
```

Comment:
Using a map here would reduce redundancy and make the block more readable.

Reply:
hm, generally yes, but its not so trivial. Some of the values are double and some bool, so at least two maps. Furthermore, I would need to create pointers for all paramters.